slider/replay.py

```diff
@@ -152,6 +154,64 @@ def _consume_actions(buffer):
     return out
 
 
+def _ms(t):
+    return t.total_seconds() * 1e3
+
+
+def _within(p1, p2, d):
+    return (p1.x - p2.x) ** 2 + (p1.y - p2.y) ** 2 < d ** 2
+
+
+def _pressed(datum):
+    return datum.key1 or datum.key2 or datum.mouse1 or datum.mouse2
+
+
+def _process_circle(obj, rdatum, hw, scores):
+    out_by = abs(_ms(rdatum.offset - obj.time))
+    if out_by < hw.hit_300:
+        scores["perfects"].append(obj)
+    elif out_by < hw.hit_100:
+        scores["goods"].append(obj)
+    elif out_by < hw.hit_50:
```

Comment:
It (should be) impossible to reach since this function is only reached when a hit is detected and hits are only looked for within the 50 hit window. So yes it is a miss but it's covered earlier on in the main function